tests/console/wicked_basic.pm

```diff
@@ -46,7 +47,18 @@ sub run {
     type_string("#***Test 3: Bring down the wicked server service***\n");
     systemctl('stop wickedd.service');
     assert_wicked_state(wicked_daemon_down => 1);
-    assert_script_run('! ifdown $(ls -d /sys/class/net/!(lo) | head -1 | sed "s/.*\///")');
+    assert_script_run("! ifdown $iface");
+    type_string("#***Test 4: Bring up the wicked server service***\n");
+    systemctl('start wickedd.service');
+    assert_wicked_state();
+    assert_script_run("ifup $iface");
+    type_string("#***Test 5: List the network interfaces with wicked***\n");
+    my @wicked_all_ifaces = split("\n", script_output("wicked show --brief all"));
+    foreach (@wicked_all_ifaces) {
+        $_ = substr($_, 0, index($_, ' '));
+    }
+    my @ls_all_ifaces = split(' ', script_output("ls /sys/class/net/"));
+    die "Wrong list of interfaces from wicked" if @wicked_all_ifaces ~~ @ls_all_ifaces;
```

Comment:
Take a look at this link, what you want to do is to compute a difference, please add a function for this :) as it might be useful in the future... and maybe with @DrMullings 's hackweek project it could be documented somewhere

https://docstore.mik.ua/orelly/perl4/cook/ch04_10.htm